src/lib.rs

```diff
@@ -153,6 +153,19 @@ impl KernelConfig {
         }
     }
 
+    /// Add a set of capabilities.
+    ///
+    /// On success returns Ok, else return 1 when capabilities you provided are not all supported by kernel.
+    pub fn add_capabilities(
+        self: &mut Self,
+        capabilities_to_add: u32) -> Result<(), u16> {
+        if capabilities_to_add & self.capabilities != capabilities_to_add {
+            return Err(1);
```

Comment:
Instead of returning `1` on error, can you return the set of requested capabilities that can't be supported? i.e. `capabilities_to_add - (capabilities_to_add & self.capabilities)`. I think that will be more useful to people who need to handle the error